### PR TITLE
aws- add tag/remove tag action to kinesis-video

### LINKED ISSUE #8449: aws - kinesisvideo tagging action
### Describe the bug

1) Looks like resourcegroupstagging is not supported for kinesis-video - We could fall back on the service boto3 apis for [tagging](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/kinesisvideo/client/tag_stream.html)
```
>>> response = client.tag_resources(ResourceARNList=['arn:aws:kinesisvideo:us-east-1:644160558196:stream/test/1680202478363'], Tags={'c7n': 'test'})
>>> response
{'FailedResourcesMap': {'arn:aws:kinesisvideo:us-east-1:644160558196:stream/test/1680202478363': {'StatusCode': 400, 'ErrorCode': 'InvalidParameterException', 'ErrorMessage': 'Unrecognized service or resource type for tagging'}}, 'ResponseMetadata': {'RequestId': '804b8286-4a5e-4bf4-8678-6b329bfd53ca', 'HTTPStatusCode': 200, 'HTTPHeaders': {'x-amzn-requestid': '804b8286-4a5e-4bf4-8678-6b329bfd53ca', 'content-type': 'application/x-amz-json-1.1', 'content-length': '222', 'date': 'Thu, 30 Mar 2023 19:13:22 GMT'}, 'RetryAttempts': 0}}
```

2) The retry function currently targets `LimitExceededException` instead of `ClientLimitExceededException`. We're getting throttled on ListStreams
```
botocore.errorfactory.ClientLimitExceededException: An error occurred (ClientLimitExceededException) when calling the ListStreams operation: Limit exceeded on number of requests for ListStreams API
```

### What did you expect to happen?

I expected 2 + 2 to equal 4, but instead 2 + 2 equaled 6!

### Cloud Provider

Amazon Web Services (AWS)

### Cloud Custodian version and dependency information

_No response_

### Policy

```shell
policies:
  - name: kinesisv2-filter
    resource: kinesis-video
    actions:
      - type: tag
        key: Owner
        value: c7n
```

### Relevant log/traceback output

```shell
2023-03-30 06:36:58,683 - custodian.actions - ERROR - Exception with tags: {'X': 'Y'}  Resource Tag Errors {'arn:aws:kinesisvideo:us-east-1:123456789123:stream/test/xyz': 'InvalidParameterException', 'arn:aws:kinesisvideo:us-east-1:123456789123:stream/test/xyz': 'InvalidParameterException', 'arn:aws:kinesisvideo:us-east-1:123456789123:stream/test/xyz'


botocore.errorfactory.ClientLimitExceededException: An error occurred (ClientLimitExceededException) when calling the ListStreams operation: Limit exceeded on number of requests for ListStreams API
```

### Extra information or context

_No response_

### PR DESCRIPTION
**Pull request by timmygrable**
Closes #8449

### CODE REVIEW

**Review by PratMis — APPROVED** [review]
lgtm, thanks

File: c7n/resources/kinesis.py [hunk 1]
```diff
@@ -397,3 +400,52 @@ def process(self, resources):
...
+                actions:
+                  - type: tag
+                    key: KinesisVideoTag
+                    value: "KinesisVideo Tag Value"
+    """
+    permissions = ('kinesisvideo:TagResource',)
+    
+    def process_resource_set(self, client, resource_set, tag_keys):
+        for r in resource_set:
+            client.tag_resource(ResourceARN=r['StreamARN'], Tags=tag_keys)
```
Comment (kapilt) [inline c7n/resources/kinesis.py:426]: ideally this would try/except catch not found

**Review by kapilt — APPROVED** [review]
lgtm, one minor

**Comment by timmygrable** [comment]
Hi @kapilt, thanks for the review! I added the try/catch, but can't seem to figure out how to pass codecov for the `except` block.

**Pull request merged**